components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAOTest.java

```diff
@@ -0,0 +1,1518 @@
+
+    private CacheBackedIdPMgtDAO cacheBackedIdPMgtDAO;
+    private IdPManagementDAO idPManagementDAO;
+
+    private void initiateH2Database(String databaseName, String scriptPath) throws Exception {
+
+        BasicDataSource dataSource = new BasicDataSource();
+        dataSource.setDriverClassName("org.h2.Driver");
+        dataSource.setUsername("username");
+        dataSource.setPassword("password");
+        dataSource.setUrl("jdbc:h2:mem:test" + databaseName);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.createStatement().executeUpdate("RUNSCRIPT FROM '" + scriptPath + "'");
+        }
+        dataSourceMap.put(databaseName, dataSource);
+    }
+
+    private static String getFilePath(String fileName) {
+
+        if (StringUtils.isNotBlank(fileName)) {
+            return Paths.get(System.getProperty("user.dir"), "src", "test", "resources", "dbscripts", fileName)
+                    .toString();
+        }
+        throw new IllegalArgumentException("DB Script file name cannot be empty");
+    }
+
+    private static void closeH2Database() throws Exception {
+
+        BasicDataSource dataSource = dataSourceMap.get(DB_NAME);
+        if (dataSource != null) {
+            dataSource.close();
+        }
+    }
+
+    private static Connection getConnection(String database) throws SQLException {
+
+        if (dataSourceMap.get(database) != null) {
+            return dataSourceMap.get(database).getConnection();
+        }
+        throw new RuntimeException("No datasource initiated for database: " + database);
+    }
+
+    @BeforeMethod
+    public void setup() throws Exception {
+
+        idPManagementDAO = new IdPManagementDAO();
+        cacheBackedIdPMgtDAO = new CacheBackedIdPMgtDAO(idPManagementDAO);
+        initiateH2Database(DB_NAME, getFilePath("h2.sql"));
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getTenantDomain(anyInt())).thenReturn(TENANT_DOMAIN);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        closeH2Database();
+    }
+
+    @DataProvider
+    public Object[][] getIdPsData() {
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, TENANT_DOMAIN, 2},
+                {SAMPLE_TENANT_ID2, TENANT_DOMAIN, 1},
+                {4, TENANT_DOMAIN, 0}
+        };
+    }
+
+    @Test(dataProvider = "getIdPsData")
+    public void testGetIdPs(int tenantId, String tenantDomain, int idpCount) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+
+        try (Connection connection = getConnection(DB_NAME)) {
+
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+
+            List<IdentityProvider> idps1 = cacheBackedIdPMgtDAO.getIdPs(connection, tenantId, tenantDomain);
```

Comment:
Change variable name to more meaning full one. Suggestion idpList1

Reply:
Meaningful variable names added